### PR TITLE
refactor(preboot): turn on more strict check options about tsconfig.json

### PR DESCRIPTION
**Pull request by vvakame**
* **Please check if the PR fulfills these requirements**
- [X] The commit message follows our guidelines: https://github.com/angular/preboot/blob/master/CONTRIBUTING.md#commit-message-format
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

* **What modules are related to this pull-request**
- [ ] server side
- [ ] client side
- [ ] inline
- [X] build process
- [ ] docs
- [ ] tests

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

improve `preboot.d.ts` and other ts files.
Now, preboot (and angular2-platform-node) are not compatible with `--noImplicitAny` options.
This is spread affect to user land tsconfig settings.
I want to use `--noImplicitAny` options in my project.

* **What is the current behavior?** (You can also link to an open issue here)

Users can't use `--noImplicitAny` with this package.

* **What is the new behavior (if this is a feature change)?**

We can use `--noImplicitAny` with this package!

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

maybe no.

* **Other information**:

None.

### CODE REVIEW

File: src/browser/preboot_browser.ts [hunk 1]
```diff
@@ -22,13 +22,13 @@ import {
 } from '../preboot_interfaces';
 
 // the idea here is that preboot is a global value on the window that is used by the client
-var preboot = prebootClient();
```
Comment (gdi2290) [inline src/browser/preboot_browser.ts]: is the build system exposing this variable at all or was this unnecessary to begin with?

Comment (vvakame) [inline src/browser/preboot_browser.ts]: you are right. preboot variable is required.
I think this ts file has a es2015 module syntax, this variable is not required.
But it is wrong. import statement is removed at compile time.

`<script src="preboot_client.min.js"></script>`
In this case, this variable is visible from global namespace.

**Comment by vvakame** [comment]
updated. please review again 😸

**Comment by gdi2290** [comment]
LGTM

**Comment by vvakame** [comment]
👀

**Pull request merged**